packages/perseus/src/perseus-types.ts

```diff
@@ -11,72 +11,89 @@ export type Size = [number, number];
 export type CollinearTuple = [vec.Vector2, vec.Vector2];
 export type ShowSolutions = "all" | "selected" | "none";
 
+/**
+ * Our core set of Perseus widgets.
+ *
+ * This interface is the basis for "registering" all Perseus widget types.
+ * There should be one key/value pair for each supported widget. If you create
+ * a new widget, an entry should be added to this interface. Note that this
+ * only registers the widget options type, you'll also need to register the
+ * widget so that it's available at runtime (@see
+ * {@link file://./widgets.ts#registerWidget}).
+ *
+ * Importantly, the key is not important and is not used anywhere outside of
+ * this interface so it is arbitrary.
+ *
+ * If you define the widget outside of this package, you can still add the new
+ * widget to this interface by writing the following in that pacakge that
+ * contains the widget. TypeScript will merge that definition of the
+ * `PerseusWidgets` with the one defined below.
+ *
+ * ```typescript
+ * declare module "@khanacademy/perseus" {
+ *     interface PerseusWidgetTypes {
+ *         Awesomeness: MyAwesomeNewWidget;
+ *     }
+ * }
+ *
+ * type MyAwesomeNewWidget = WidgetOptions<'awesomeness', MyAwesomeNewWidgetOptions>;
+ * ```
+ *
+ * This interface can be extended through the magic of TypeScript "Declaration
+ * merging". Specifically, we augment this module and extend this interface.
+ *
+ * @see {@link https://www.typescriptlang.org/docs/handbook/declaration-merging.html#module-augmentation}
+ */
+export interface PerseusWidgetTypes {
+    Categorizer: CategorizerWidget;
+    CSProgram: CSProgramWidget;
+    Definition: DefinitionWidget;
+    Dropdown: DropdownWidget;
+    Explanation: ExplanationWidget;
+    Expression: ExpressionWidget;
+    Grapher: GrapherWidget;
+    GradedGroupSet: GradedGroupSetWidget;
+    GradedGroup: GradedGroupWidget;
+    Group: GroupWidget;
+    IFrame: IFrameWidget;
+    Image: ImageWidget;
+    InputNumber: InputNumberWidget;
+    Interaction: InteractionWidget;
+    InteractiveGraph: InteractiveGraphWidget;
+    LabelImage: LabelImageWidget;
+    Matcher: MatcherWidget;
+    Matrix: MatrixWidget;
+    Measurer: MeasurerWidget;
+    MoleculeRenderer: MoleculeRendererWidget;
+    NumberLine: NumberLineWidget;
+    NumericInput: NumericInputWidget;
+    Orderer: OrdererWidget;
+    PassageRefTarget: RefTargetWidget;
+    PassageRef: PassageRefWidget;
+    Passage: PassageWidget;
+    PhetSimulation: PhetSimulationWidget;
+    Python: PythonProgramWidget;
+    Plotter: PlotterWidget;
+    Radio: RadioWidget;
+    Sorter: SorterWidget;
+    Table: TableWidget;
+    Video: VideoWidget;
```

Comment:
Not related to your PR and no action needed: can we leverage this or something like this to get rid of `basic-widgets`, `extra-widgets`, and `all-widgets`?

Reply:
I don't think so.  This interface is a registry of widget option types whereas the things you mentioned are the "runtime registries".  We could certainly merge all those three items together though.